layersvt/api_dump.h

```diff
@@ -475,43 +475,69 @@ class ApiDumpSettings {
 
         vkuSetLayerSettingCompatibilityNamespace(layerSettingSet, GetDefaultPrefix());
 
+        // Read the format type first as it may be used in the output file extension
+        output_format = ApiDumpFormat::Text;
+        if (vkuHasLayerSetting(layerSettingSet, kSettingsKeyOutputFormat)) {
+            std::string value;
+            vkuGetLayerSettingValue(layerSettingSet, kSettingsKeyOutputFormat, value);
+            value = ToLowerString(value);
+            if (value == "html") {
+                output_format = ApiDumpFormat::Html;
+            } else if (value == "json") {
+                output_format = ApiDumpFormat::Json;
+            } else {
+                output_format = ApiDumpFormat::Text;
+            }
+        }
+
         // If the layer settings file has a flag indicating to output to a file,
-        // do so, to the appropriate filename.
+        // do so, to the appropriate default filename.
         std::string filename_string = "";
         if (vkuHasLayerSetting(layerSettingSet, kSettingsKeyFile)) {
             bool file = false;
             vkuGetLayerSettingValue(layerSettingSet, kSettingsKeyFile, file);
 
-            if (file) {
-                if (vkuHasLayerSetting(layerSettingSet, kSettingsKeyLogFilename)) {
-                    vkuGetLayerSettingValue(layerSettingSet, kSettingsKeyLogFilename, filename_string);
-                    if (filename_string.empty()) {
-                        filename_string = "vk_apidump.txt";
-                    }
-                }
+            if (output_format == ApiDumpFormat::Html) {
+                filename_string = "vk_apidump.txt";
+            } else if (output_format == ApiDumpFormat::Json) {
+                filename_string = "vk_apidump.txt";
```

Comment:
Should all three of these be the same?  Should html end with .html?

Reply:
Thats a copy paste error! Good catch.